src/plots/cartesian/autorange.js

```diff
@@ -0,0 +1,324 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var isNumeric = require('fast-isnumeric');
+
+var Lib = require('../../lib');
+var FP_SAFE = require('../../constants/numerical').FP_SAFE;
+
+module.exports = {
+    getAutoRange: getAutoRange,
+    makePadFn: makePadFn,
+    doAutoRange: doAutoRange,
+    needsAutorange: needsAutorange,
+    expand: expand
+};
```

Comment:
I initially pulled this module out of `axes.js` to avoid circular references; then I ended up moving `clearCalc` into `setConvert` instead of here, which mooted the circular reference problem, but I still like having these separate from the behemoth `axes.js`.